ipatests/test_integration/test_http_kdc_proxy.py

```diff
@@ -4,53 +4,202 @@
+        kdc_url = 'https://{}/KdcProxy'.format(self.master.hostname)
+        kdc_option = 'kdc = {}'.format(kdc_url)
+
+        # configure kdc proxy for IPA realm
+        krb5conf, n = re.subn(r' kdc = .+', kdc_option, krb5conf)
+        assert n == 1
+
+        # configure kdc proxy for Windows AD realm
+        ad_realm_config = '''
+        {realm} = {{
+            {kdc}
+        }}
+        '''.format(realm=self.ad.domain.realm, kdc=kdc_option)
+        krb5conf, n = re.subn(
+            '\[realms\]', '[realms]' + ad_realm_config, krb5conf)
+        assert n == 1
+
+        self.client.put_file_contents(paths.KRB5_CONF, krb5conf)
+        self.client.run_command(['systemctl', 'restart', 'sssd.service'])
+        yield
+        krb5conf_backup.restore()
+        self.client.run_command(['systemctl', 'restart', 'sssd.service'])
+
+    @contextmanager
+    def configure_kdc_proxy_for_ad_trust(self, use_tcp):
+        backup = tasks.FileBackup(self.master, paths.KDCPROXY_CONFIG)
+        with tasks.remote_ini_file(self.master, paths.KDCPROXY_CONFIG) as conf:
+            conf.set('global', 'use_dns', 'true')
+            conf.set('global', 'configs', 'mit')
+            if use_tcp:
+                conf.add_section(self.ad.domain.realm)
+                conf.set(self.ad.domain.realm, 'kerberos',
+                         'kerberos+tcp://{}:88'.format(self.ad.hostname))
+                conf.set(self.ad.domain.realm, 'kpasswd',
+                         'kpasswd+tcp://{}:464'.format(self.ad.hostname))
+        try:
+            self.master.run_command(['ipactl', 'restart'])
+            yield
+        finally:
+            backup.restore()
+            self.master.run_command(['ipactl', 'restart'])
+
+    @pytest.mark.parametrize('user_origin', ['ipa', 'ad'])
+    def test_user_login_on_client_without_firewall(self, users, user_origin):
+        """Basic check for test setup."""
+        tasks.clear_sssd_cache(self.master)
+        user = users[user_origin]
+        tasks.kinit_as_user(self.client, user['name'], user['password'])
+
+    @pytest.mark.usefixtures('restrict_network_for_client')
+    @pytest.mark.parametrize('user_origin', ['ipa', 'ad'])
+    def test_access_blocked_on_client_without_kdcproxy(
+            self, users, user_origin):
+        """Check for test firewall setup."""
+        tasks.clear_sssd_cache(self.master)
+        user = users[user_origin]
+        result = tasks.kinit_as_user(
+            self.client, user['name'], user['password'], raiseonerr=False)
+        expected_error = (
+            "Cannot contact any KDC for realm '{}' while getting initial "
+            "credentials".format(user['domain'].realm))
+        assert result.returncode == 1 and expected_error in result.stderr_text
+
+    @pytest.mark.usefixtures('restrict_network_for_client',
+                             'client_use_kdcproxy')
+    def test_ipa_user_login_on_client_with_kdcproxy(self, users):
```

Comment:
Please use follow black's and PEP-8's formatting style:
```

    @pytest.mark.usefixtures(
        'restrict_network_for_client',
        'client_use_kdcproxy'
    )
    def test_ipa_user_login_on_client_with_kdcproxy(self, users):
```

Reply:
Could you please point which pep-8 style rule you are referring to? 
And https://www.freeipa.org/page/Python_Coding_Style does not seem to mention any other rules...

Reply:
It's black code style, https://github.com/psf/black#the-black-code-style

Reply:
I tried to use black utility with new code and it heavily reformatted the code, not only the mentioned lines. One of the changes is making lines longer then 79 chars which violates current  code style.
But even with "-l 79" the changes are huge.
So could you please tell me, is the use of black formater now agreed and enforced for the FreeIPA project? I see it is not mentioned in https://www.freeipa.org/page/Python_Coding_Style, neither it is being checked by PR-CI.